src/askui_runner/modules/core/infrastructure/files/askui.py

```diff
@@ -104,136 +86,121 @@ def sync(
+                        remote_file.url, local_path, remote_file.last_modified, dry
                     )
                 else:
-                    action_mapper["skip"](relative_path)
+                    logging.info(f"Skip {relative_path} (no changes)")
                     continue
 
             # File exists only locally
             elif local_file:
                 if source_of_truth == "local":
-                    action_mapper["upload"](local_file.path, remote_file_path)
+                    self._upload_file(local_file.path, remote_file_path, dry)
                 elif delete:
-                    action_mapper["delete_local"](local_file.path)
+                    self._delete_local_file(local_file.path, dry)
 
             # File exists only remotely
             elif remote_file:
                 if source_of_truth == "remote":
                     local_path = os.path.join(local_dir_path, relative_path)
-                    action_mapper["download"](
-                        remote_file.url, remote_file.last_modified, local_path
+                    self._download_file(
+                        remote_file.url, local_path, remote_file.last_modified, dry
                     )
                 elif delete:
-                    action_mapper["delete_remote"](remote_file_path)
+                    self._delete_remote_file(remote_file_path, dry)
 
-    # region Private Functions
     @retry(stop=stop_after_attempt(5), wait=wait_exponential(), reraise=True)
-    def _upload_file(self, local_file_path: str, remote_file_path: str) -> None:
+    def _upload_file(
+        self, local_file_path: str, remote_file_path: str, dry=False
+    ) -> None:
+        url = urljoin(
+            base=self._base_url + "/",
+            url=quote(remote_file_path),
+        )
+
+        if dry:
+            logging.info(f"Dry: Uploading {local_file_path} to {url}.")
+            return
+
         with open(local_file_path, "rb") as f:
-            url = urljoin(
-                base=self.base_url + "/",
-                url=remote_file_path,
-            )
-            url = quote(url, safe=":/")
             with requests.put(
                 url,
                 files={"file": f},
-                headers=self.headers,
+                headers=self._headers,
                 timeout=UPLOAD_REQUEST_TIMEOUT_IN_S,
                 stream=True,
             ) as response:
                 if response.status_code != 200:
                     response.raise_for_status()
 
     @retry(stop=stop_after_attempt(5), wait=wait_exponential(), reraise=True)
-    def _delete_file(self, remote_file_path: str) -> None:
-        delete_url = urljoin(self.base_url + "/", remote_file_path)
-        response = requests.delete(delete_url, headers=self.headers)
+    def _delete_remote_file(self, remote_file_path: str, dry=False) -> None:
+        if dry:
+            logging.info(f"Dry: Deleting file {remote_file_path}")
+            return
```

Comment:
What about always logging it and depending on `dry` option executing the logic or not.